flow/project.py

```diff
@@ -1685,18 +1685,86 @@ def _store_bundled(self, operations):
         else:
             h = '.'.join(op.id for op in operations)
             bid = '{}/bundle/{}'.format(self, sha1(h.encode('utf-8')).hexdigest())
-            fn_bundle = self._fn_bundle(bid)
+            fn_bundle = self._fn_stored('.bundles', bid)
             os.makedirs(os.path.dirname(fn_bundle), exist_ok=True)
             with open(fn_bundle, 'w') as file:
                 for operation in operations:
                     file.write(operation.id + '\n')
             return bid
 
+    def _store_aggregates(self, operations):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param operations:
+            The operations to be submitted.
+        :type operations:
+            A sequence of instances of :py:class:`.JobOperation`
+        """
+        for operation in operations:
+            aid = '{}.txt'.format(operation.name)
+            aggregate_wid = '{} {} \n'.format(operation.id, ' '.join(map(str, operation._jobs)))
+            fn_aggregate = self._fn_stored('.aggregates', aid)
+            try:
+                os.makedirs(os.path.dirname(fn_aggregate), exist_ok=True)
+            except PermissionError:
+                continue
+            if os.path.exists(fn_aggregate):
+                with open(fn_aggregate, 'r') as file:
+                    if aggregate_wid in file:
+                        continue
+            with open(fn_aggregate, 'a') as file:
+                file.write(aggregate_wid)
+
+    def _fetch_aggregates(self, group):
+        """Store aggregate-ids per operation information.
+
+        This enables status check of aggregates which were
+        formed previously but are not present currently for any
+        operation.
+
+        :param group:
+            FlowGroup associated with the operation.
+        :type group:
+            list :py:class:`flow.FlowGroup`
+        :return:
+            Fetched signac job handles that were previously submitted.
+        :rtype:
+            list
+        """
+        fetched_aggregates = []
+        dir = '.aggregates/{}.txt'.format(group.name)
+        if os.path.exists(dir):
+            with open(dir, 'r') as file:
+                for obj in file:
+                    fetched_aggregate = []
+                    _ids = obj.split(' ')
+                    submission_id = _ids[0]
+                    job_ids = _ids[1:-1]
+                    try:
+                        for job_id in job_ids:
+                            fetched_aggregate.append(self.open_job(id=job_id))
+
+                        # Checking whether the aggregate and the submission id match.
+                        # If not, then a user must have changed the submission id.
+                        # Hence skip this aggregate.
+                        assert group._generate_id(fetched_aggregate) == submission_id
```

Comment:
Users can change the submission ids (which they shouldn't) but there's a possibility that they change the id, hence we need to check this

Reply:
Can they change submission ids without accessing the file directly or using internal APIs? If not, I wouldn't bother checking.

Reply:
They can't do that using internal API but there's a probability that they add some jobs manually into that file. We'd then display the status info of those jobs (Which technically we're not supposed to do)
The `id` is only stored to look after edge cases like these.

Reply:
If a user messes with a file in `.aggregates` that is on them, in my opinion.

Reply:
I'm not sure how the last comment on this thread relates to the remainder of it. Adding the submission id is not related to this check being performed... right?

Reply:
So when someone submits an operation, we stored the information in a hidden folder. If we store submission id then the user won't be able to edit those ids (even willingly, as they're all unique) but if we just store the aggregate ids, then users might be able to add different aggregate ids by themselves which I believed may create some unnecessary errors. But I get @b-butler 's point that it should be on them if they mess up with the hidden file.

Moreover, this thread is now outdated as this execution is shifted to the next PR because actual aggregation was to be involved for this.